### ISSUE TITLE
Cannot create japanese name channel in v4.5

### ISSUE THREAD

**Issue by kaakaa** [mattermost/mattermost-server#8001]
#### Summary
In Mattermost v4.5, I cannot create japanese name channel (e.g. `日本語`) .
**However, By [PR-464](https://github.com/mattermost/mattermost-webapp/pull/464) this issues looks to be fixed in my local environment.**

Do I have to wait for v4.6 for this fix to be applied?
If **yes**, please close this issue. I'll wait Mattermost v4.6.

---

For more details.

In mattermost 4.4, specifying Japanese as the display name of the New Channel modal and pressing the `Create New Channel` button, the`Set Channel URL` modal was open. (Because japanese  word cannot use as channel url)
However, when doing the same operation after updating to mm4.5, `Set Channel URL` modal don't open. (Nothing is displayed on the browser)

#### Steps to reproduce
1. Open `New Channel` modal
2. Input Japanese word in `Name`
3. Push `Create New Channel` button

<img width="720" alt="2017-12-23 16 13 05" src="https://user-images.githubusercontent.com/1453749/34318266-ef07d588-e806-11e7-8bc5-c949d6042753.png">

#### Expected behavior
Open `Set Channel URL` modal

#### Observed behavior (that appears unintentional) 
Nothing to display

<img width="720" alt="2017-12-23 16 13 55" src="https://user-images.githubusercontent.com/1453749/34318261-e39c3522-e806-11e7-95c5-54ba1889a281.png">

#### Possible fixes
It looks that the error occurs at https://github.com/mattermost/mattermost-webapp/blob/release-4.5/components/change_url_modal.jsx#L42.

I think it's caused by the following breaking changes of react v16.0.

[React v16.0 - React Blog](https://reactjs.org/blog/2017/09/26/react-v16.0.html)
> Shallow renderer no longer calls componentDidUpdate because DOM refs are not available. This also makes it consistent with componentDidMount (which does not get called in previous versions either).

**Comment by Bloodoff** [comment]
Same error with cyrillic channel name, mattermost version 4.5.0

**Comment by lindy65** [comment]
Hi @kaakaa @Bloodoff 

Thanks for your feedback,

I have reproduced your issue on our nightly build server running off latest master so I believe there is an issue even although [this PR](https://github.com/mattermost/mattermost-webapp/pull/464) has been merged.  I get the error `Name must be 2 or more lowercase alphanumeric characters`.

I have created [this ticket](https://mattermost.atlassian.net/browse/PLT-8454) so we can look into the issue.
